File: 283.移动零.cs
/*
 * @lc app=leetcode.cn id=283 lang=csharp
 *
 * [283] 移动零
 */

// @lc code=start
public partial class Solution
{
    public void MoveZeroes(int[] nums)
    {
        int count = 0;
        int len = nums.Length;
        for (int i = 0; i < len; i++)
        {
            int num = nums[i];
            if (num == 0)
            {
                count++;
            }
            else
            {
                nums[i - count] = num;
            }
        }
        for (int i = len - 1; i > len - 1 - count; i--)
        {
            nums[i] = 0;
        }
    }
}
// @lc code=end

File: src/283.移动零.rs
/*
 * @lc app=leetcode.cn id=283 lang=rust
 *
 * [283] 移动零
 */

// @lc code=start
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {
        let mut zero_index = 0;
        let len = nums.len();
        for fast_index in 0..len {
            let num = nums[fast_index];
            if num != 0 {
                nums[fast_index] = 0;
                nums[zero_index] = num;
                zero_index += 1;
            }
        }
    }
}
// @lc code=end
